### PR TITLE
Enhancement/control flow pleeease

### PR DESCRIPTION
**Pull request by gloopydoop**
here comes the linting....

### CODE REVIEW

File: sources/mma/mma.f90 [hunk 1]
```diff
@@ -78,7 +82,8 @@ module mma
 
      !> Interface for updating the MMA
      generic, public :: update => mma_update_cpu, mma_update_vector
-     procedure, pass(this) :: mma_update_cpu
+     ! for debugging I'm making this public
```
Comment (timfelle) [inline sources/mma/mma.f90]: Are you done debugging so we can revert this?

Comment (gloopydoop) [reply sources/mma/mma.f90]: The dimension are still a bit awkward, but I haven't written the `optimizer_t` so making this public was quickest solution

**Review by gloopydoop — COMMENTED** [review]
ok now I'm done.

File: sources/mma/mma.f90 (hunk 1 above)
Comment (gloopydoop) [reply sources/mma/mma.f90]: switched back

File: examples/rugby_ball/woop
Comment (timfelle) [inline examples/rugby_ball/woop:1]: Please name scripts for executing examples `run.sh` such that the scripting system can utilize it.

File: sources/mma/mma.f90 [hunk 2]
```diff
@@ -50,7 +50,11 @@ module mma
      private
 
      real(kind=rp) :: a0, f0val, asyinit, asyincr, asydecr, epsimin, &
-          residumax, residunorm
+          residumax
+     ! we need to have the KKT residual public
+     ! or alternatively, have KKT a function not a subroutine,
+     ! that return residunrom
+     real(kind=rp), public :: residunorm
```
Comment (timfelle) [inline sources/mma/mma.f90]: This is an internal variable and should never be accessed from the outside. Why did you need this?

File: sources/topology_optimization/bkp_design.f90 [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+  !> @brief Pointer to the design domain mask on the device
+  type(c_ptr) :: design_domain_mask_d = c_null_ptr
+
+  ! ========================================================================= !
+  ! Topology type
+  ! ========================================================================= !
+
+  !> @brief The topology type, which is used to describe the designs in
+  !! topology optimization
+  type, extends(simulation_component_t) :: design_t
```
Comment (timfelle) [inline sources/topology_optimization/bkp_design.f90]: This should not be a simulation component.

File: sources/topology_optimization/design.f90 [hunk 4]
```diff
@@ -42,31 +56,43 @@ module design_module
 
   !> @brief The topology type, which is used to describe the designs in
   !! topology optimization
-  type :: design_t
+  type, extends(simulation_component_t) :: design_t
```
Comment (timfelle) [inline sources/topology_optimization/design.f90:59]: `design_t` have been defined both here and in `sources/bkp_design.f90` That will not be possible to compile.

File: visualisation/debug_top.pvsm
Comment (timfelle) [inline visualisation/debug_top.pvsm:1]: Why are there changes to the debug paraview file?

**Review by timfelle — CHANGES_REQUESTED** [review]

File: examples/rugby_ball/woop
Comment (gloopydoop) [reply examples/rugby_ball/woop:1]: we can just delete that it's not being used anymore

File: sources/mma/mma.f90 (hunk 2 above)
Comment (gloopydoop) [reply sources/mma/mma.f90]: replaced with get_residunorm

File: sources/topology_optimization/bkp_design.f90 (hunk 3 above)
Comment (gloopydoop) [reply sources/topology_optimization/bkp_design.f90]: that was a backup of the pre-existing design that was already there. we can delete it.

File: visualisation/debug_top.pvsm
Comment (gloopydoop) [reply visualisation/debug_top.pvsm:1]: I didn't touch those...

File: sources/mma/mma.f90 [hunk 5]
```diff
@@ -50,7 +50,8 @@ module mma
      private
 
      real(kind=rp) :: a0, f0val, asyinit, asyincr, asydecr, epsimin, &
-          residumax, residunorm
+          residumax
+     real(kind=rp) :: residunorm
```
Comment (timfelle) [inline sources/mma/mma.f90]:
```suggestion
          residumax, residunorm
```
Let's not do empty changes to files

File: sources/neko_ext/neko_ext.f90 [hunk 6]
```diff
@@ -54,7 +54,9 @@ subroutine reset(neko_case)
     v => neko_case%fluid%v
     w => neko_case%fluid%w
     p => neko_case%fluid%p
+    if(allocated(neko_case%scalar)) then
     s => neko_case%scalar%s
+    end if
```
Comment (timfelle) [inline sources/neko_ext/neko_ext.f90]: Missing indentation

File: visualisation/debug_top.pvsm
Comment (timfelle) [reply visualisation/debug_top.pvsm:1]: Then they should not be in the PR.

File: cmake/build_example.cmake
Comment (timfelle) [inline cmake/build_example.cmake:1]: These changes do not belong in this PR, they should be done seperately

File: .vscode/tasks.json
Comment (timfelle) [inline .vscode/tasks.json:1]: Same with this one, It is not related to the controlflow.

File: examples/debug/debug.case
Comment (timfelle) [inline examples/debug/debug.case:1]: This should probably not be modified in this PR should it?

File: examples/rugby_ball/topopt.f90
Comment (timfelle) [inline examples/rugby_ball/topopt.f90:1]: This file is not listed in the CMakeLists, so why is it part of the PR

File: examples/rugby_ball/bkp_driver.f90
Comment (timfelle) [inline examples/rugby_ball/bkp_driver.f90:1]: This file is not used in the example, please delete it

File: examples/rugby_ball/new_design.f90
Comment (timfelle) [inline examples/rugby_ball/new_design.f90:1]: This file is not part of the example and should be deleted from the PR

**Review by timfelle — CHANGES_REQUESTED** [review]

File: sources/neko_ext/neko_ext.f90 (hunk 6 above)
Comment (gloopydoop) [reply sources/neko_ext/neko_ext.f90]: done

File: examples/rugby_ball/topopt.f90
Comment (gloopydoop) [reply examples/rugby_ball/topopt.f90:1]: done

File: examples/rugby_ball/bkp_driver.f90
Comment (gloopydoop) [reply examples/rugby_ball/bkp_driver.f90:1]: done

File: examples/rugby_ball/new_design.f90
Comment (gloopydoop) [reply examples/rugby_ball/new_design.f90:1]: done

**Review by timfelle — APPROVED** [review]
